docs(process-engine): add description for composite incident handlers (#1079)
camunda/camunda-docs-manual: content/user-guide/process-engine/incidents.md

Please be aware that this API is **not** part of the [public API]({{< ref "/introduction/public-api.md" >}}) and may change in later releases.

{{< /note >}}

# Composite Incident Handlers

By default, an incident can only be handled by one handler with a same type.

> Composite incident handler contains main and subs handlers. Only the result from the "main" incident handler will be returned.

Reviewer: ```suggestion
Composite incident handlers allow defining one main and multiple sub handlers. Only the result from the "main" incident handler will be returned.
```

> To enable composite handlers configure property:

Reviewer: ```suggestion
To enable composite incident handlers, configure the following property:
```

```xml
<property name="compositeIncidentHandlersEnabled" value="true" />
```

> If you want to provide a multiple incident handler`s you can add one or multiple incident handlers using the following method:

Reviewer: ```suggestion
If you want to provide multiple incident handlers, you can add them using the following method:
```

```java
org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl.setCustomIncidentHandlers(List<IncidentHandler>)
```

> All customIncidentHandlers will be added as sub handlers to `CompositeIncidentHandler` for same handler type.

Reviewer: ```suggestion
All additional incident handlers will be added as sub handlers to the `CompositeIncidentHandler` for the same handler type.
```

> By default, main handler is `DefaultIncidentHandler`. To override the main handler you need create `CompositeIncidentHandler` with your main `IncidentHandler` and init incidentHandlers before setting up the engine.

Reviewer: ```suggestion
By default, the main handler is `DefaultIncidentHandler`. To override the main handler, create a `CompositeIncidentHandler` with your own main `IncidentHandler` and initialize the incident handlers in the engine configuration before setting up the engine.
```
